lib/core.ps1

```diff
@@ -823,17 +823,21 @@ function is_scoop_outdated() {
 }
 
 function substitute($entity, [Hashtable] $params, [Bool]$regexEscape = $false) {
-    if ($entity -is [Array]) {
-        return $entity | ForEach-Object { substitute $_ $params $regexEscape}
-    } elseif ($entity -is [String]) {
+    if ($entity -is [String]) {
         $params.GetEnumerator() | ForEach-Object {
-            if($regexEscape -eq $false -or $null -eq $_.Value) {
+            if ($regexEscape -eq $false -or $null -eq $_.Value) {
                 $entity = $entity.Replace($_.Name, $_.Value)
             } else {
                 $entity = $entity.Replace($_.Name, [Regex]::Escape($_.Value))
             }
         }
         return $entity
+    } elseif ($entity -is [Array]) {
+        return $entity | ForEach-Object { substitute $_ $params $regexEscape }
+    } elseif ($entity -is [PSObject]) {
+        $newentity = $entity.PSObject.Copy()
+        $newentity.PSObject.Properties.ForEach( { $_.Value = substitute $_.Value $params $regexEscape })
```

Comment:
Why `.ForEach` instead of `| ForEach-Object`

![A](https://i.imgur.com/qbUvvlH.png)

![B](https://i.imgur.com/0FR5fVb.png)